ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -198,7 +198,10 @@ def fetch_sample_data(self, schema: str, table: str) -> Optional[TableData]:
 
     def next_record(self) -> Iterable[Entity]:
         inspector = inspect(self.engine)
-        schema_names = inspector.get_schema_names()
+        if self.config.database and self.config.scheme == "trino":
+            schema_names = [self.config.database]
+        else:
+            schema_names = inspector.get_schema_names()
```

Comment:
I'm not sure that this applies to other connectors. `database` in Trino is a container or `tables` not `schemas`.

Reply:
Hi @amiorin, Thanks for the PR. It is better to override the the next_record in [trino.py](https://github.com/open-metadata/OpenMetadata/blob/ccb96b3ea7b3452854be6eba11e611d355ea1b58/ingestion/src/metadata/ingestion/source/trino.py#L106) to keep the sql_source more generalised 

also as suggested by harsha we can move the initialisation of schema_names to prepare method as it is required to be executed only once. I have a similar PR open #3489 for mysql, you can refer that.

cc: @ayush-shah @harshach

Reply:
Done